### PR TITLE
Reservation forms: Email & sms

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-215
https://reload.atlassian.net/browse/DDFSOEG-210

#### Description

Adds reservation forms for changing email and telephone number.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

Changed the useButtonHandler hook so it now supports both modal open and close.

### CODE REVIEW

**Review by kasperbirch1 — APPROVED** [review]

File: src/components/atoms/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { FC } from "react";
```
Comment (Adamik10) [inline src/components/atoms/TextInput.tsx:1]: Very nice addition, this component :) 
Consider placing this file in a folder in the components folder that is called Input.

File: src/components/reservation/forms/ReservationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        )}
+        <div className="reservation-form__footer mt-48">
+          <Button
+            label="Gem"
+            buttonType="none"
+            disabled={false}
+            collapsible={false}
+            size="xlarge"
+            variant="filled"
+            elementType="submit"
```
Comment (Adamik10) [inline src/components/reservation/forms/ReservationForm.tsx:37]: remove elementType pls :)

File: src/components/reservation/forms/helper.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  type,
+  changedText,
+  savedText,
+  patron,
+  mutate,
+  queryClient
+}: SaveText) => {
+  // If we do not have an email address, we do not want to save anything.
+  if (!changedText) {
+    return;
```
Comment (Adamik10) [inline src/components/reservation/forms/helper.ts:84]: Consider adding a // TODO: xyz comment here about validation.
Or a comment by the form or submit button.

File: src/components/reservation/forms/helper.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    patron
+  });
+
+  // If cannot construct the updated patron data or the email address is the same,
+  // we do not want to save anything.
+  if (!updatedPatronData || !textDiffers) {
+    return;
+  }
+
+  // Update user email.
```
Comment (Adamik10) [inline src/components/reservation/forms/helper.ts:99]:
```suggestion
  // Update user data.
```

**Review by Adamik10 — APPROVED** [review]
Great stuff :) I have a couple of comments that I left :) Please have a look at those.

File: src/apps/material/material.entry.tsx [hunk 5]
```diff
@@ -79,6 +79,14 @@ interface MaterialEntryTextProps {
   librariesHaveTheMaterialText: string;
   findOnShelfModalScreenReaderModalDescriptionText: string;
   findOnShelfModalCloseModalAriaLabelText: string;
+  modalReservationFormEmailHeaderTitleText: string;
```
[the remark this answers is not in the document]
Comment (spaceo) [reply src/apps/material/material.entry.tsx:82]: 👍

File: src/components/Buttons/Button.tsx [hunk 6]
```diff
@@ -14,6 +14,7 @@ export type ButtonProps = {
   onClick?: () => void;
   classNames?: string;
   id?: string;
+  elementType?: "button" | "submit";
```
[the remark this answers is not in the document]
Comment (spaceo) [reply src/components/Buttons/Button.tsx:17]: 👍

File: src/components/atoms/TextInput.tsx (hunk 1 above)
Comment (spaceo) [reply src/components/atoms/TextInput.tsx:1]: Good idea ;)

File: src/components/reservation/forms/helper.ts (hunk 4 above)
Comment (spaceo) [reply src/components/reservation/forms/helper.ts]: fixed ;)

File: src/components/reservation/forms/ReservationForm.tsx (hunk 2 above)
Comment (spaceo) [reply src/components/reservation/forms/ReservationForm.tsx]: Jason DONEnowich

**Review by sinejespersen — APPROVED** [review]

**Pull request merged**